### ISSUE TITLE
Buble don't exclude node_modules

### ISSUE THREAD

**Issue by csvwolf** [JeffreyWay/laravel-elixir-webpack-official#20]
Error Info: 

```
Module build failed: SyntaxError: 'with' in strict mode (1784:36)
    at Parser.pp$4.raise (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:2221:15)
    at Parser.pp$1.parseWithStatement (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:932:27)
    at Parser.pp$1.parseStatement (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:708:32)
    at Parser.pp$1.parseBlock (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:981:25)
    at Parser.pp$3.parseFunctionBody (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:2105:24)

    at Parser.pp$1.parseFunction (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:1065:10)
    at Parser.pp$3.parseExprAtom (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:1810:19)
    at Parser.parseExprAtom (/Users/SkyAo/Self_Develop/blog/node_modules/buble/dist/buble.umd.js:656:26)
    at Parser.pp$3.parseExprSubscripts (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/Users/SkyAo/Self_Develop/blog/node_modules/acorn/dist/acorn.js:1692:19)
 @ ./resources/assets/js/app.js 16:0-35
```

package.json:

```
{
  "private": true,
  "scripts": {
    "prod": "gulp --production",
    "dev": "gulp watch"
  },
  "devDependencies": {
    "babel-core": "^6.0.0",
    "babel-loader": "^6.0.0",
    "babel-plugin-transform-runtime": "^6.15.0",
    "babel-preset-es2015": "^6.14.0",
    "bootstrap-sass": "^3.3.7",
    "cross-env": "^1.0.6",
    "css-loader": "^0.23.1",
    "file-loader": "^0.8.5",
    "gulp": "^3.9.1",
    "jquery": "^3.1.0",
    "laravel-elixir": "^6.0.0-9",
    "laravel-elixir-vue-2": "^0.2.0",
    "laravel-elixir-webpack-official": "^1.0.2",
    "lodash": "^4.14.0",
    "style-loader": "^0.13.1",
    "vue-loader": "^9.4.2"
  },
  "dependencies": {
    "element-ui": "^1.0.0-rc.4",
    "vue": "^2.0.0-rc.6",
    "vue-resource": "^1.0.3"
  }
}
```

gulpfile:

``` javascript
const elixir = require('laravel-elixir');
const path = require('path');

require('laravel-elixir-vue-2');
require('laravel-elixir-webpack-official');

Elixir.webpack.mergeConfig({
  resolveLoader: {
    root: path.join(__dirname, 'node_modules'),
  },
  module: {
    loaders: [
      {
        test: /\.js$/,
        loader: 'babel',
        exclude: /node_modules/
      },
      {
        test: /\.css$/,
        loader: 'style!css'
      },
      {
        test: /\.(eot|svg|ttf|woff|woff2)$/,
        loader: 'file'
      },
      {
        test: /\.(png|jpg|gif|svg)$/,
        loader: 'file',
        query: {
          name: '[name].[ext]?[hash]'
        }
      }
    ]
  }
});

elixir(mix => {
    mix.sass('app.scss')
       .webpack('app.js');
});
```

Although I want to use my webpack config to merge it, I found the buble still running from the error info.

Can I replace `buble-loader` to `babel-loader` or make a configure for `buble-loader`? 

[The same problem when use the Vue Component](https://github.com/ElemeFE/element/issues/160)

**Comment by davericher** [comment]
Do not include in webpack official
and put this

```
Elixir.webpack.mergeConfig({
  resolveLoader: {
    root: path.join(__dirname, 'node_modules'),
  },
  module: {
    loaders: [
      {
        test: /\.js$/,
        loader: 'babel',
        exclude: /node_modules/
      },
      {
        test: /\.css$/,
        loader: 'style!css'
      },
      {
        test: /\.(eot|svg|ttf|woff|woff2)$/,
        loader: 'file'
      },
      {
        test: /\.(png|jpg|gif|svg)$/,
        loader: 'file',
        query: {
          name: '[name].[ext]?[hash]'
        }
      }
    ]
  }
});
```

inside the elixir callback.

**Comment by csvwolf** [comment]
I remove the webpack official and then put the code inside the elixir callback:

``` javascript
elixir(mix => {
    Elixir.webpack.mergeConfig({
        resolveLoader: {
            root: path.join(__dirname, 'node_modules'),
        },
        module: {
            loaders: [
            {
                test: /\.js$/,
                loader: 'babel',
                exclude: /node_modules/
            },
            {
                test: /\.css$/,
                loader: 'style!css'
            },
            {
                test: /\.(eot|svg|ttf|woff|woff2)$/,
                loader: 'file'
            },
            {
                test: /\.(png|jpg|gif|svg)$/,
                loader: 'file',
                query: {
                name: '[name].[ext]?[hash]'
                }
            }
            ]
        }
    });
    mix.sass('app.scss')
       .webpack('app.js');
});
```

But it doesn't work.

**Comment by JeffreyWay** [comment]
Can you confirm that this is still an issue, even for version 1.0.8 of this package?

**Comment by csvwolf** [comment]
Well, my current packages:

```
{
  "private": true,
  "scripts": {
    "prod": "gulp --production",
    "dev": "gulp watch"
  },
  "devDependencies": {
    "bootstrap-sass": "^3.3.7",
    "css-loader": "^0.25.0",
    "element-ui": "^1.0.0-rc.5",
    "gulp": "^3.9.1",
    "jquery": "^3.1.0",
    "laravel-elixir": "^6.0.0-9",
    "laravel-elixir-vue-2": "^0.2.0",
    "laravel-elixir-webpack-official": "^1.0.8",
    "lodash": "^4.16.2",
    "style-loader": "^0.13.1",
    "vue": "^2.0.1",
    "vue-loader": "^9.5.1",
    "vue-resource": "^1.0.3"
  },
  "dependencies": {
    "element-ui": "^1.0.0-rc.4"
  }
}
```

It doesn't work. The same error info.

**Comment by webcodetailor** [comment]
Here is an example of the working configuration: https://github.com/ElementUI/element-in-laravel-starter/blob/master/gulpfile.js

The important thing here is to reset default module loaders with this line `Elixir.webpack.config.module.loaders = [];`

Also you don't have to include 'laravel-elixir-webpack-official' in your gulp file but then you should put this line `Elixir.webpack.config.module.loaders = [];` inside of `elixir(mix => { ... });`

You could also use _webpack.config.js_ file instead of mergin webpack configs right in your gulp file.

**Comment by csvwolf** [comment]
@webcodetailor Yes, the file is written by me. But I think the implementation is very ugly since it overwrites the whole default loader configurations.

**Comment by borger** [comment]
@JeffreyWay it does happen on v1.0.8. I'm having similar issues trying to use [vue-icons](https://github.com/vue-comps/vue-icons). The error is very similar:

```
Module build failed: SyntaxError: 'with' in strict mode (181:37)
    at Parser.pp$4.raise (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:2221:15)
    at Parser.pp$1.parseWithStatement (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:932:27)
    at Parser.pp$1.parseStatement (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:708:32)
    at Parser.pp$1.parseBlock (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:981:25)
    at Parser.pp$3.parseFunctionBody (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:2105:24)
    at Parser.pp$1.parseFunction (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:1065:10)
    at Parser.pp$3.parseExprAtom (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:1810:19)
    at Parser.parseExprAtom (/Sites/vue-icons/node_modules/buble/dist/buble.umd.js:656:26)
    at Parser.pp$3.parseExprSubscripts (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/Sites/vue-icons/node_modules/buble/node_modules/acorn/dist/acorn.js:1692:19)
 @ ./~/vue-icons/icon-font-compatible.js 4:6-23
```

**Comment by targit86** [comment]
I'm getting the same error with both `babel-loader` and `buble-loader`. Webpack is dying when trying to import either of the [vue-recaptcha](https://www.npmjs.com/package/vue-recaptcha) and [vue-multiselect](https://www.npmjs.com/package/vue-multiselect) components.

FWIW, you can get around the issue with `vue-multiselect` by importing the Vue file directly from its node_modules directory; i.e. `import MultiSelect from 'vue-multiselect/lib/Multiselect.vue';` instead of `import MultiSelect from 'vue-multiselect';`. This workaround does _not_ work with `vue-recaptcha`.

Here's my `webpack.config.js`:

``` javascript
const path = require('path');

module.exports = {
    resolveLoader: {
        root: path.join(__dirname, 'node_modules'),
    },
    module: {
        loaders: [
            {
                test   : /\.js$/,
                include: path.join(__dirname, 'resources/assets'),
                exclude: path.join(__dirname, 'node_modules'),
                loader : 'babel',
            },
            {
                test  : /\.vue$/,
                loader: 'vue',
            }
        ]
    },
};
```

**Edit**: here's a link to the same issue from the `vue-multiselect` repo: https://github.com/monterail/vue-multiselect/issues/124

**Comment by JeanLucEsser** [comment]
I'm having the exact same issue with vue-infinite-loading.
I'm not able to find any workaround.

Even tried to use rollup in place of webpack with no results.

Any update on the issue?

**Comment by iraklisg** [comment]
@JeffreyWay similar issue when I try to  `npm install --save vue-material` using `laravel-elixir-webpack-official@1.0.8` and a fresh install of Laravel v5.3.19 framework.

**_In my assets/js/app.js entry file_**

```
require('./bootstrap');

// Import vue-material
import VueMaterial from 'vue-material'
// import 'vue-material/dist/vue-material.css'

// Enable Vue Material in my application
Vue.use(VueMaterial);

// To get Vue Material working properly, need to configure and apply a default theme.
Vue.material.theme.register('default', {
  primary: 'cyan',
  accent: 'pink'
})


const app = new Vue({
    el: '#app',
   /* ... */
}); 
```

**_In my gulpfile.js_**:

```
const elixir = require('laravel-elixir');

require('laravel-elixir-vue-2');


elixir(mix => {
    mix.sass('app.scss')
        .webpack('app.js');
});
```

When running `gulp` I get the following error: 

```
{ [Error: ./~/vue-material/dist/vue-material.js
Module build failed: SyntaxError: 'with' in strict mode (8:13503)
    at Parser.pp$4.raise (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:2221:15)
    at Parser.pp$1.parseWithStatement (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:932:27)
    at Parser.pp$1.parseStatement (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:708:32)
    at Parser.pp$1.parseBlock (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:981:25)
    at Parser.pp$3.parseFunctionBody (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:2105:24)
    at Parser.pp$1.parseFunction (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:1065:10)
    at Parser.pp$3.parseExprAtom (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:1810:19)
    at Parser.parseExprAtom (/home/igeorgas/webApps/vue/node_modules/buble/dist/buble.umd.js:656:26)
    at Parser.pp$3.parseExprSubscripts (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:1715:21)
    at Parser.pp$3.parseMaybeUnary (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:1692:19)
 @ ./resources/assets/js/app.js 10:0-38]
  message: './~/vue-material/dist/vue-material.js\nModule build failed: SyntaxError: \'with\' in strict mode (8:13503)\n    at Parser.pp$4.raise (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:2221:15)\n    at Parser.pp$1.parseWithStatement (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:932:27)\n    at Parser.pp$1.parseStatement (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:708:32)\n    at Parser.pp$1.parseBlock (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:981:25)\n    at Parser.pp$3.parseFunctionBody (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:2105:24)\n    at Parser.pp$1.parseFunction (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:1065:10)\n    at Parser.pp$3.parseExprAtom (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:1810:19)\n    at Parser.parseExprAtom (/home/igeorgas/webApps/vue/node_modules/buble/dist/buble.umd.js:656:26)\n    at Parser.pp$3.parseExprSubscripts (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:1715:21)\n    at Parser.pp$3.parseMaybeUnary (/home/igeorgas/webApps/vue/node_modules/acorn/dist/acorn.js:1692:19)\n @ ./resources/assets/js/app.js 10:0-38',
  showStack: false,
  showProperties: true,
  plugin: 'webpack-stream',
  __safety: { toString: [Function: bound ] } }
```

**Comment by csvwolf** [comment]
The pull request solved the problem in 1.0.9 now. #32
